src/game/updateGame.js

```diff
@@ -0,0 +1,21 @@
+export const updateGame = (
+  showAlert,
+  closeAlert,
+  cellValue,
+  boardData,
+  currentPlayer,
+  changeTurn,
+  updateBoard
+) => {
+  let updatedBoard = boardData;
+
+  if (cellValue !== "X" && cellValue !== "O") {
+    closeAlert();
+    updatedBoard = updateBoard(boardData, cellValue, currentPlayer);
+    changeTurn();
+  } else {
+    showAlert();
+  }
```

Comment:
To continue my previous Q, perhaps you are passing in show/close alert as a way to "do something when there is an invalid state". If so, I wonder if you can decouple the concept of showing the alert from the concept of an invalid state. An initial idea is to rename the callbacks so they are not based on the UI, so if the UI changes later to show something else (instead of an alert), your interface is the same. ex: onValidationFailure() is a name of a function that I can think of, preliminarily. If this doesn't make sense for what you were trying to achieve, no worries, just ignore this comment.

Reply:
One more thing to consider, I see that updateGame happens inside <Board/>. Is this the optimal place for it? Would board control how the game is played in addition to how it is rendered? 
This isn't a deal breaker for me for PR approval, but of all the comments I added, I think this is the one that I would like you to see if you can find an alternative solution to. It seems like Board is doing too much. Can you think of how data flows in your app and what could be missing here?

Reply:
Also in functional programming, the pattern is to not mutate a value; ie "let" is avoided. Can you see how to refactor it to use const instead?

Reply:
Awesome. This is good advice. I'll work on moving the updateGame functionality out of Board and into a Game component that calls Board. I think this would make a lot more sense and maybe minimize the amount of props being passed into Board.

I definitely see what you mean about showAlert/closeAlert. I'll refactor this into a validator of some kind. Thank you!